### DISCUSSION TITLE
Need help with troubleshooting: Gitlab pages and pelican themes

### DISCUSSION THREAD

**Discussion by FranAyrolo** [getpelican/pelican#3138] [Q&A] [answered]
Hello everyone.
I recently started putting together a simple website, hosting it on Gitlab pages. I managed to get the pipeline going and it properly deploys the site, and the 2 base themes do actually work.
Now... locally it works™️ . Not every theme does, but I found dev-random2 to be a nice looking one and something I can work up from, so that's the one I'm going for. I can properly run pelican and generate the site, navigate through it and all, so I know the problem is somewhere between Gitlab and my pipeline. Within Gitlab's pipeline, the 'pages' and 'pages:deploy' jobs both pass correctly, so I'm even more at a loss.
To add even more confusion to the mix, part of the theme DOES get carried into the page! The original theme is in French, which I translated locally in my PC into Spanish. When loading the site, I can see all the text-related things the theme adds, in French, like the headers for articles and the sidebar entries, but it's all in plain text, in a single column.
I don't expect a miraculous answer, but I've been googling around for why this happens and I still can't figure it out. I need help simply knowing where to look for the error. I'm sorry if this is really basic stuff, I'm just a simple C/C++ dev
My relevant config files are in a pastebin, here:  https://pastebin.com/fGs2DfrN
And also the log from the Gitlab job. I don't have access to the "pages:deploy" job, only the "pages" one: https://pastebin.com/0WCdUdnQ
Here's a link to the site, too. It's still just different test articles to get the hang of it. [franayrolo.gitlab.io](https://franayrolo.gitlab.io/)
If there's a better way of sharing that, please let me know. My working environment is Linux 21.1 Vera, and I visualize the page with Firefox, but my roommate uses Chromium browser and showed me it doesn't load properly on his either.

**Comment by avaris** [comment, marked as answer]
Theme resources are not retrieved due to mixed protocol and that's because of your `SITEURL='franayrolo.gitlab.io'`. That's interpreted as `http://franayrolo.gitlab.io` while gitlab serves the page with `https`, and that's typically blocked by browsers since requesting unsecure content in a secure site is usually a bad idea.  So you need to adjust it to `SITEURL='https://franayrolo.gitlab.io'` or the protocol relative version `SITEURL='//franayrolo.gitlab.io'`.

Next, your workflow installs the version from `pelican-themes`, which will pull the original (French) not your local version. And don't use `pelican-themes --install` (we really should deprecate that). Just place your theme folder next to your config and specify the folder name in your config (i.e. `THEME = 'my-dev-random2'`)

**Reply by FranAyrolo** [reply]
Thanks for the reply!
Ok, mixed results hehe. The theme folder worked, I uninstalled the theme and just dropped a theme/dev-random2 folder in there, and added it as you said to the conf. When deploying the page, things are now translated.
However, it's still all plain text, no theme. 
Regarding your first answer, about the https siteurl, it was actually commented away to begin with. I tried doing SITEURL='https://franayrolo.gitlab.io', but it breaks the deployed site. As soon as you navigate into anything within the page, it add the site's URL again into the browser's URL (so you get, for example "`https://franayrolo.gitlab.io/franayrolo.gitlab.io/index.html`". It also breaks locally, since I can load the localhost:8000 address just fine, but as soon as I navigate into anything, it breaks and goes plain.
Does this have anything to do with the RELATIVE_URLS variable (which I'm not touching, btw)

**Reply by avaris** [reply]
> However, it's still all plain text, no theme.

Yes, that's because browser does not load CSS due to mixed protocol. That's the issue. [*]

> Regarding your first answer, about the https siteurl, it was actually commented away to begin with

I'm guessing here but deployed site uses `publishconf.py`, so I'd suggest checking in there :).

> Does this have anything to do with the RELATIVE_URLS variable (which I'm not touching, btw)

Yes and no. It will make all urls relative to the file. It will "fix" things since you'll no longer link with absolute urls but it's a different kind of fix. And regarding "localhost", yes giving specifying `SITEURL` will break that. It is expected. That's why there is 2 config files: `pelicanconf.py` and `publishconf.py`. Keep `SITEURL=''` in your `pelicanconf.py` for local testing, but specify correct `SITEURL` in `publishconf.py` so that deployed site will generate proper links.

<hr>

PS: [*] Actually, I think I'm slightly wrong on that. Your css links are like this:

```
<link rel="stylesheet" type="text/css" href="franayrolo.gitlab.io/theme/css/theme.css">
```

Which is relative and resolved to `https:///franayrolo.gitlab.io/franayrolo.gitlab.io/theme/css/theme.css`, and that's obviously wrong and does not exist. That triggers a redirect to signup page on gitlab and that's when cross site policy is triggered. But solution is the same regardless.

**Reply by FranAyrolo** [reply]
Ahhh, I see.
Well, that fixed it! Left pelicanconf with '' and publish with the //site... and it did the trick. Thank you so much :)
When you say 

> It will make all urls relative to the file. It will "fix" things since you'll no longer link with absolute urls

What that means is that, say, the about page is relative to where it is in my directory structure, from the base folder which is content. Is that correct?

**Reply by avaris** [reply]
It is relative to the file that link is in. Assuming an output folder structure like this:
```
|- page1.html
|- folder1
    |- page2.html
|- folder2
    |- page3.html
    |- page4.html
```

`page1.html` will link to `page4.html` with `href="folder2/page4.html"`
`page2.html` will link to `page4.html` with `href="../folder2/page4.html"`
`page3.html` will link to `page4.html` with `href="page4.html"`   (or `href="./page4.html"` rather)

without relative urls (a.k.a. absolute urls), a link to `page4.html` will be `href="http://example.com/folder2/page4.html"` irrespective of the file that does the linking.

**Reply by FranAyrolo** [reply]
Ahh, that makes sense. I can't tell which would be better, but I'd rather not fix what isn't broken (yet)
Thank you for the replies :)

**Reply by avaris** [reply]
It's been long debated whether relative urls are fine or not for search engine optimization (SEO) practices. It's been argued that relative urls can lead to seemingly different links pointing to the same page and that's something to avoid, so favored absolute urls for "better to be safe". I don't know what the current consensus is, but it's probably more of a personal preference nowadays :).
